python/ray/_private/runtime_env/pip.py

```diff
@@ -209,9 +206,16 @@ def _install_pip_packages(
         python = _PathHelper.get_virtualenv_python(path)
         # TODO(fyrestone): Support -i, --no-deps, --no-cache-dir, ...
         pip_requirements_file = _PathHelper.get_requirements_file(path)
-        with open(pip_requirements_file, "w") as file:
-            for line in pip_packages:
-                file.write(line + "\n")
+
+        def _gen_requirements_txt():
+            with open(pip_requirements_file, "w") as file:
+                for line in pip_packages:
+                    file.write(line + "\n")
+
+        # Avoid blocking the event loop.
+        loop = asyncio.get_running_loop()
```

Comment:
Maybe add TODO here, when ray drop py3.6, we modify it to `asyncio.to_thread`

Reply:
The `asyncio.to_thread` was new in Python 3.9. It's a long time to drop Python 3.8.